libs/SmartStore/SmartStore/Classes/SFSmartStore.m

```diff
@@ -1351,6 +1341,19 @@ - (void)registerSoupWithSpec:(SFSoupSpec*)soupSpec withIndexSpecs:(NSArray*)inde
     }
     [self insertIntoSoupIndexMap:soupIndexMapInserts withDb:db];
     
+    // LogEvent
+    NSMutableArray<NSString *> *features = [[NSMutableArray alloc] init];
+    if (soupUsesJSON1) {
+        [features addObject:@"JSON1"];
+    }
+    if (soupUsesExternalStorage) {
+        [features addObject:@"ExternalStorage"];
+    }
+    if ([self hasFts:soupSpec.soupName withDb:db]) {
```

Comment:
Instead, we could define the following (right after defining soupUsesJSON1):
```objective-c
BOOL soupUsesFts = [SFSoupIndex hasFts:indexSpecs];
```
Then there is no need to move the block of code.